### ISSUE TITLE
`Hash`, `Dictionary` and `Set`

### ISSUE THREAD

**Issue by WilsonGramer** [wipplelang/wipple#57]
Implement the following APIs:

```wipple
Hash : A => trait (A -> Number)
dictionary :: Key Value where (Hash Key) => ... -> Dictionary Key Value
set :: Element where (Hash Element) => ... -> Set Element
```

**Comment by WilsonGramer** [comment]
`Hash` and `Dictionary` implemented in 065116c!

**Comment by WilsonGramer** [comment]
`Set` implemented in `0a939fb`!
